(#1898) Adds External Link Icon to Feature Cards

dlescarbeau opened the pull request:
## Pull Request Details

UPDATES SASS VERSION TO 1.69.0 AS A PEER DEPENDENCY

- Adds mixin for adding an external link icon to external links
- Utilizes new mixin to add external link icon to feature cards
- Updates `nci-card` twig file to fix an issue with its alt-text
- Updates `nci-card-group` twig file to accommodate the `modifier` that can be passed to `nci-card`

Closes #1898>

## Author PR Checklist
Items that the author of the PR is responsible for checking before submitted the PR.

### General:
- [x] I have reviewed the acceptance criteria defined in the ticket and ensured the work has been completed.
- [x] The commit message passes [all quality commit message standards](https://github.com/NCIOCPL/ncids/wiki/Creating-Quality-Commit-Messages).
- [x] Unit tests have been updated or created to reflect any javascript changes.
- [x] Storybook scenarios have been updated or created to reflect any html/css/js changes.

### Accessibility:
- [ ] WCAG 2.1 Level AA requirements have been met.

### Development:
- [x] Any new or updated javascript code has 100% unit test coverage.
- [x] New or updated breakpoints have regression images.
- [x] Breaking changes have been thoroughly documented in the PR.

## Product Reviewer PR Checklist
Items the product team is responsible for reviewing.

### General:
- [x] There are no unexpected or unapproved regression image changes. 
- [x] Functionality of interactive elements meet the acceptance criteria.
- [x] The product is visually and functionally the same across the different browsers.

### Accessibility:
- [x] AxeDev Tools: there are no new or outstanding accessibility issues introduced in this PR.
- [x] Lighthouse: scores have not noticeably decreased during this PR.
- [x] Wave: there are no new errors or contrast errors introduced in this PR.

## Design Reviewer PR Checklist
Items the design team is responsible for reviewing.  

### General:
- [ ] New or updated features introduced in this PR are developed mobile-first.
- [ ] Breakpoint changes and regression images match those breakpoints.
- [ ] This PR has been tested in all supported browsers at all breakpoints.

## Developer Reviewer PR Checklist
Items the development team is responsible for reviewing.

### General:
- [ ] New code passes code quality standards set by industry standards.
- [ ] The expected Storybook stories have been added or updated for the new or updated feature.
- [ ] The expected unit tests have been added or updated for the new or updated feature.

### Accessibility:
- [ ] VoiceOver: Described content matches with what was expected.
- [ ] Keyboard navigation: new or updated features and content are navigable via the keyboard.

andyvanavery31 commented:
Notes/Changes Requested:
- All three of the External Link Icon examples will be shown on the Documentation Site (with External Link, with External Link Override to Not Show Icon, and Internal Link Overridden to Show Icon)
- Rename the Code Snippets to "Card with External Link", "Card with Internal Link with External Link Override", and "Card with External Link with Icon Removed Override"
- Update the code snippets to use the content from the Scenarios section of the ticket to reduce user confusion.
- Update storybook story from "External Cards" to "Cards with External Links"

dlescarbeau commented:
> Notes/Changes Requested:
> 
> * All three of the External Link Icon examples will be shown on the Documentation Site (with External Link, with External Link Override to Not Show Icon, and Internal Link Overridden to Show Icon)
> * Rename the Code Snippets to "Card with External Link", "Card with Internal Link with External Link Override", and "Card with External Link with Icon Removed Override"
> * Update the code snippets to use the content from the Scenarios section of the ticket to reduce user confusion.
> * Update storybook story from "External Cards" to "Cards with External Links"

- updated titles of code snippets per above to be "Card with External Link", "Card with Internal Link with External Icon Override", and "Card with External Link with Icon Removed Override"
- updated code snippet content
- updated story in storybook to be "Cards with External Links"

bryanpizzillo commented on packages/ncids-css/packages/nci-card/internal/_nci-card.scss:
```diff
@@ -1,9 +1,33 @@
 @use '../../../internal-uswds/usa-card/src/styles/usa-card' as card;
 
+// Mixin to pass for overriding certain external icon styles
+@mixin external-icon() {
```
can you make these mixins something like "nci-card-external-icon" with the component name? Just so we don't get confused...

dlescarbeau replied:
I thought it being a mixin only used inside `_nci-card.scss` made that pretty obvious, but sure, whatever's best.

dlescarbeau commented on patches/@uswds__uswds@3.7.1.patch:
```diff
@@ -1278,6 +1278,165 @@ index 983d2ed46d758291a903957e79df24c9043260dd..40a1f924922b3dd40129c28e453e51ea
+ 
+-$external-link-size: 1.75ex;
+-
+-$icon-object: (
+-  "name": "launch",
+-  "color": currentColor,
+-  "height": $external-link-size,
+-  "svg-height": 24,
+-  "svg-width": 24,
+-  "position-x": center,
+-  "position-y": center,
+-);
```
sure

adrianacastaneda commented:
IA approval from both me and Lindsay! Thank you!

blilianyu commented:
Looks great thanks!

bryanpizzillo commented on packages/ncids-css/package.json:
```diff
@@ -42,7 +42,7 @@
 		"postcss-loader": "6.1.1",
 		"rimraf": "~3.0.2",
-		"sass": "~1.49.0",
+		"sass": "~1.69.0",
```
We need to also set a peerDependency since you actually need to have 1.69 for this to work.

bryanpizzillo reviewed (approved):
Looks good!
